### PR TITLE
Update plugin version and remove powermock

### PR DESCRIPTION
**Pull request by aykborstelmann**
Because of #7, I tried to fix the build. 
Turns out, that the error in the build was because the intellij gradle plugin was too old. 
When updating this it turned out that powermock didn't work. Since I don't like the usage of Powermock anyway, I removed it.

### CODE REVIEW

**Comment by aykborstelmann** [comment]
@cowclaw could your review this when you've got time ?

**Comment by aykborstelmann** [comment]
To avoid, or at least get notifed when this plugin doesn't build anymore I suggest running a monthly github actions build a09dafd29be06b7a24408d9aebc22a44cc024aca

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: CI
 
 on:
+  schedule:
+    - cron: '0 0 1 * *'
```
Comment (dsp-de) [inline .github/workflows/gradle.yml:5]: This runs the build on the first day of the month. If this is your intention, that is fine. If you want to run it at 1 AM, take '0 0 1 * * *'.

File: build.gradle [hunk 2]
```diff
@@ -35,7 +35,11 @@ patchPluginXml {
 }
 
 runPluginVerifier {
-	ideVersions.set(["IU-193.7288.26", "IU-203.8084.24", "IU-211.7628.15"])
+	ideVersions.set([
+			"IU-203.8084.24",
+			"IU-211.7628.15",
+			"IU-222.4167.21"
```
Comment (dsp-de) [inline build.gradle]: Maybe update the most recent one to the most recent release: "IU-222.4167.29"

File: build.gradle [hunk 3]
```diff
@@ -45,7 +45,5 @@ dependencies {
 	testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.5.1'
 	testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-params', version: '5.3.2'
 	testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.13.2'
-	testImplementation group: 'org.powermock', name: 'powermock-module-junit4', version: '2.0.2'
-	testImplementation group: 'org.powermock', name: 'powermock-api-mockito2', version: '2.0.2'
 	testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.0.0'
```
Comment (dsp-de) [inline build.gradle]: mockito-core 3.0.0 failed on my machine with adopt-openjdk-16 (language level 11) with error message "Mockito cannot mock this class". Changing my JDK would be an appropriate solution. Nevertheless, mockito-core 4.8.0 works with this JDK, so if there are no other obstacles, you may consider using the latest version.

File: build.gradle [hunk 4]
```diff
@@ -45,7 +45,5 @@ dependencies {
 	testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.5.1'
```
Comment (dsp-de) [inline build.gradle]: Comment on the first two dependencies:

"testImplementation group: 'junit', name: 'junit', version: '4.12'" has a vulnerability:
"CVE-2020-15250 5.5 Incorrect Permission Assignment for Critical Resource vulnerability pending CVSS allocation"
https://www.cvedetails.com/cve/CVE-2020-15250
Suggestion: The dependency does not seem to be necessary. What about deleting it?

"testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.4.2'"
could be "testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.9.0'"

Same for the other two dependencies of the same group.

AssertJ could be 3.23.1

File: src/test/java/de/cronn/validation_files_diff/ValidationDiffTest.java [hunk 5]
```diff
@@ -154,30 +151,27 @@ public void testGetDirDiffElementFromPath() {
...
+		doReturn(fileSystem).when(virtualFileManager).getFileSystem(LocalFileSystem.PROTOCOL);
+		application.registerService(VirtualFileManager.class, virtualFileManager);
 
-		PowerMockito.when(ModuleManager.getInstance(any())).thenAnswer(invocation -> moduleManager);
-		PowerMockito.when(ValidationDiffApplicationOptionsProvider.getInstance()).thenAnswer(invocation -> applicationSettings);
-		PowerMockito.when(ValidationDiffProjectOptionsProvider.getInstance(eq(project))).thenAnswer(invocation -> projectSettings);
-		PowerMockito.when(DirDiffManager.getInstance(eq(project))).thenAnswer(invocation -> dirDiffManager);
-		PowerMockito.when(LocalFileSystem.getInstance()).thenAnswer(invocation -> fileSystem);
+		project.registerService(ValidationDiffProjectOptionsProvider.class, new ValidationDiffProjectOptionsProviderImpl());
+		project.registerService(DirDiffManager.class, dirDiffManager);
```
Comment (dsp-de) [inline src/test/java/de/cronn/validation_files_diff/ValidationDiffTest.java:171]: Nice to see that getting rid of the static mocking brings you closer to the real setup. :+1:

**Review by dsp-de — COMMENTED** [review]
Nice catching up with the recent versions. The mock replacements are an improvement.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (aykborstelmann) [reply .github/workflows/gradle.yml:5]: Yes this was my intention

File: build.gradle (hunk 2 above)
Comment (aykborstelmann) [reply build.gradle]: Done

File: build.gradle (hunk 4 above)
Comment (aykborstelmann) [reply build.gradle]: Done (by you)

File: build.gradle (hunk 3 above)
Comment (aykborstelmann) [reply build.gradle]: Done (by you)

File: src/test/java/de/cronn/validation_files_diff/ValidationDiffTest.java (hunk 5 above)
Comment (aykborstelmann) [reply src/test/java/de/cronn/validation_files_diff/ValidationDiffTest.java:171]: :)

**Pull request merged**